### PR TITLE
chore: Optimize/10528 remove generated svgs

### LINKED ISSUE #10528: Remove generated SVGs in contract-metadata
### What is this about?

Referencing the findings of the JS bundle here - https://www.notion.so/metamask-consensys/Performance-Game-Plan-83ec0133962a4b7682a64c66e65d566c?pvs=4#5f9c6b1abaa148a7804e6986bf98fe6c, we can see that the generated images consume ~20% of the JS bundle size. The generation was initially done to standardize the asset types for consistent rendering, however, is not absolutely necessary. Therefore, we will be removing the SVG generation along with scripts associated with it. The end result should be a 20% reduction in JS bundle size, which should improve post-app start times. This can be measured using the numbers from this ticket - https://github.com/MetaMask/metamask-mobile/issues/10527
JS bundle load time can be measured from this ticket - https://github.com/MetaMask/metamask-mobile/issues/10521

### Scenario

_No response_

### Design

_No response_

### Technical Details

- Remove scripts associated with generating SVGs for contract-metadata
- Remove SVG if they are commited
- Update codebase where it references the generated SVGs to not use the SVGs
- Verify JS bundle decreased by ~20%
- Check if there's an improvement in JS bundle load time

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- JS bundle should decrease ~20%
- Build script should no longer generate SVGs from contract-metadata
- BONUS - Check if there's an improvement in JS bundle load time

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

20% bundle size from SVG assets - https://www.notion.so/metamask-consensys/Performance-Game-Plan-83ec0133962a4b7682a64c66e65d566c?pvs=4#5f9c6b1abaa148a7804e6986bf98fe6c

### PR DESCRIPTION
**Pull request by Cal-L**
## **Description**

These changes removes the generated SVGs for contract-metadata. In addition to it taking ~7MB of the JS bundle size, it was also most likely no longer used. 

## **Related issues**

Fixes:
#10528 

## **Manual testing steps**

Navigate around where tokens appear and ensure that the tokens still render properly. Some areas to test:
- Swaps token search
- Adding tokens to wallet
- Detected tokens
- Token details

### **Before**

JS bundle size on `main`
image

### **After**

JS bundle size after removal
image

Spot check
https://github.com/user-attachments/assets/83579c20-82d9-45c0-bf9a-826cebce18df

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/AssetIcon/index.tsx [hunk 1]
```diff
@@ -7,18 +7,13 @@ import {
...
 interface Props {
   /**
-   * String of the asset icon to be searched in contractMap
+   * URL of the logo
    */
   logo: string;
-  /**
-   * Whether logo has to be fetched from @metamask/contract-metadata
-   */
-  watchedAsset?: boolean;
```
Comment (Cal-L) [inline app/components/UI/AssetIcon/index.tsx:21]: Not used

File: app/components/UI/AssetIcon/index.tsx [hunk 2]
```diff
@@ -53,11 +48,9 @@ const AssetIcon = memo((props: Props) => {
 
   const style = [styles.logo, props.customStyle];
   const isImageUrl = isUrl(props.logo) || props.logo.substr(0, 4) === 'ipfs';
-  const source: ImageSourcePropType = isImageUrl
+  const source: ImageSourcePropType | null = isImageUrl
     ? { uri: props.logo }
-    : // TODO: Replace "any" with type
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      (staticLogos as any)[props.logo];
```
Comment (Cal-L) [inline app/components/UI/AssetIcon/index.tsx:60]: This looked like dead code since I couldn't find any instance where a non url was being passed in

**Comment by Cal-L** [comment]
E2E passed on last commit with functional changes

**Comment by Cal-L** [comment]
SVGs are not commited to the repo and is generated during a release build. The SVGs will not longer be generated

**Review by tommasini — APPROVED** [review]

**Review by NicolasMassart — COMMENTED** [review]
tracking usage of this assetIcon I found app/components/UI/AssetList/index.js that doesn't seem to be used either. Dead code or mistake?

**Comment by Cal-L** [comment]
@NicolasMassart I believe the logic of contract metadata within AssetIcon is dead code, which was removed

**Pull request merged**